plugins/connection/lxd.py

```diff
@@ -70,7 +83,21 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
 
         self._display.vvv(u"EXEC {0}".format(cmd), host=self._host)
 
-        local_cmd = [self._lxc_cmd, "exec", self._host, "--", self._play_context.executable, "-c", cmd]
+        if self.get_option("project"):
+            local_cmd = [
+                self._lxc_cmd,
+                "--project", self.get_option("project"),
+                "exec",
+                "%s:%s" % (self.get_option("remote"), self._host),
+                "--",
+                self._play_context.executable, "-c", cmd]
+        else:
+            local_cmd = [
+                self._lxc_cmd,
+                "exec",
+                "%s:%s" % (self.get_option("remote"), self._host),
+                "--",
+                self._play_context.executable, "-c", cmd]
```

Comment:
This (and the other steps) could be greatly simplified (both for readability and maintanence in the future) by composing the list step for step, so that the `if` is only for adding the `--project` option.

I.e. something like
```.py
local_cmd = [self._lxc_cmd]
if self.get_option("project"):
    local_cmd.extend(["--project", self.get_option("project")])
local_cmd.extend([
    "exec",
    "%s:%s" % (self.get_option("remote"), self._host),
    "--",
    self._play_context.executable, "-c", cmd
])
```

Reply:
Thanks for the suggestion @felixfontein , I've made those changes. :)